### DISCUSSION TITLE
Branch rules applied to a Template repository

### DISCUSSION THREAD

**Discussion by ashwin-js** [community/community#110164] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

I have a template repository with some Branch protection rules . Now If I create a new repository of that template will the branch rules be still in place for the new repo as well ? If not is there any feature which currently does this?

**Comment by matej** [comment]
I've tried and branch protection rules do not carry over. This would be good to add as it would make it easier to enforce standards such as code review requirements across organizations.

**Reply by brianjmurrell** [reply]
https://github.com/orgs/community/discussions/55200

**Comment by navdotnetreqs** [comment]
It seems that
1) rulesets are not copied from template
2) organizational rulesets (for all repos) cannot be enforced unless I update to the highest enterprise plan!

No wonder they don't implement this..

**Comment by Bobston01** [comment]
When you create a new repository from a template repository on GitHub, the Branch Protection Rules do not carry over to the new repository. The new repository starts with a clean state in terms of settings like branch protection, actions, and permissions.

Why Don’t Branch Protection Rules Carry Over?
GitHub treats template repositories as a blueprint for files and structure, but repository-specific settings such as branch protection, access controls, and secrets remain unique to each repository. This is by design to prevent unintended security or workflow restrictions in the newly created repository.

Workaround to Apply Branch Rules Automatically
Currently, GitHub does not have a built-in feature to automatically transfer branch protection rules

**Comment by stupenkov** [comment]
I created a simple script for copying ruleset between repositories, it may be useful - https://github.com/stupenkov/github-ruleset

**Comment by ataylorme** [comment]
Hi all, I created https://gh-admin.com to have AI help me with this and similar things. One of the capabilities is copying rulesets from one repo to another. Here is an example prompt:

> Create the repo {org}/{repo} from the https://github.com/{org}/{repo} template. The new repo should be public. After creating the repo, copy team access, branch protection, and the .github directory from the template repo to the newly created repo.

There is a free tier, if you want to check it out. I am open to feature suggestions of other capabilities to add.
